docs/types/reference-types.rst

```diff
@@ -37,6 +37,12 @@ non-persistent area where function arguments are stored, and behaves mostly like
     data location can also be returned from functions, but it is not possible to
     allocate such types.
 
+.. note::
+    The compiler requires that ``calldata`` arrays and structs (when declared as local function
+    variables or return parameters) are initialized before use. However, in some cases with complex
+    control flow, the compiler may not detect this initialization. A common workaround in such cases
+    is to assign these variables to themselves to fulfill the compiler's requirement.
+
```

Comment:
I think the previous text was better, we just needed to adjust the bit about the workaround.

```suggestion
    Arrays and structs with ``calldata`` data location declared in a function body  
    or as its return parameters must be assigned before being used or returned.
    There are certain cases in which non-trivial control flow is used and the compiler
    can't properly detect the initialization.
    A common workaround in such cases is to assign the affected variable to itself before
    the correct initialization takes place.
```